lib/cog/chat/slack/connector.ex

```diff
@@ -323,6 +332,44 @@ defmodule Cog.Chat.Slack.Connector do
     end
   end
 
+  defp normalize_channel_id(:by_id, id = "C" <> _, slack) do
+    case slack.channels[id] do
+      nil -> classify_error(id)
+      _channel -> {:channel, id}
+    end
+  end
+  defp normalize_channel_id(:by_name, name, slack) do
+    # Private groups and channels are both noted with the '#' but they are
+    # tracked separately. So we have to merge the two lists before searching.
+    channels = slack.channels
```

Comment:
Might be more descriptive and less confusing to call this variable `channels_and_groups` instead of just `channels`